src/Microsoft.Framework.Caching.SqlServer/SqlQueries.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Caching.SqlServer
+{
+    internal class SqlQueries
+    {
+        private const string TableInfoFormat =
+            "SELECT TABLE_CATALOG, TABLE_SCHEMA, TABLE_NAME, TABLE_TYPE " +
+            "FROM INFORMATION_SCHEMA.TABLES " +
+            "WHERE TABLE_SCHEMA = '{0}' " +
+            "AND TABLE_NAME = '{1}'";
+
+        private const string GetCacheItemFormat =
+            "SELECT Id, ExpiresAtTime, SlidingExpirationInTicks, AbsoluteExpiration, Value " +
+            "FROM {0} WHERE Id = @Id AND @UtcNow <= ExpiresAtTime";
+
+        private const string GetCacheItemExpirationInfoFormat =
+            "SELECT Id, ExpiresAtTime, SlidingExpirationInTicks, AbsoluteExpiration " +
+            "FROM {0} WHERE Id = @Id AND @UtcNow <= ExpiresAtTime";
+
+        private const string SetCacheItemFormat =
+            "IF NOT EXISTS(SELECT Id FROM {0} WHERE Id = @Id) " +
```

Comment:
I wonder - could this be changed to an UPDATE statement followed by an INSERT if the `@@ROWCOUNT == 0` afterwards? In that case you'd be dropping to 1 operation instead of 2 for the update case by skipping the exists check.